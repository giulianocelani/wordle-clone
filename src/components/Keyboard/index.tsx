import Key from '../Key';

const KEYBOARD_KEYS = [
	['Q', 'W', 'E', 'R', 'T', 'Y', 'U', 'I', 'O', 'P'],
	['A', 'S', 'D', 'F', 'G', 'H', 'J', 'K', 'L'],
	['Backspace', 'Z', 'X', 'C', 'V', 'B', 'N', 'M', 'Enter']
];

type IProps = {
	onLetterPressed: (letter: string) => void;
};

const Keyboard = ({ onLetterPressed }: IProps) => {
	return (
		<div className='h-1/4 w-full flex-0 flex flex-col items-center py-5 mx-auto justify-center px-4'>
			<div className='w-full flex flex-col space-y-1 md:space-y-2 items-center'>
				{KEYBOARD_KEYS.map((row, rowIndex) => (
					<div
						key={rowIndex}
						className={`row-${
							rowIndex + 1
						} flex space-x-1 md:space-x-2 w-full items-center justify-center`}
					>
						{row.map((key, keyIndex) => (
							<Key
								key={`${rowIndex}-key-${keyIndex}`}
								letter={key}
								onClick={onLetterPressed}
							/>
						))}
					</div>
				))}
			</div>
		</div>
	);
};

export default Keyboard;
