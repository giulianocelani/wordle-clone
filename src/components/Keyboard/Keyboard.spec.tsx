import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Keyboard from '.';

describe('Keyboard', () => {
	const wrapper = shallow(<Keyboard onLetterPressed={jest.fn()} />);

	it('renders correctly', () => {
		expect(toJson(wrapper, { mode: 'deep' })).toMatchSnapshot();

		expect(wrapper.find('div.row-1').children().length).toBe(10);
		expect(wrapper.find('div.row-2').children().length).toBe(9);
		expect(wrapper.find('div.row-3').children().length).toBe(9);

		['Q', 'W', 'E', 'R', 'T', 'Y', 'U', 'I', 'O', 'P'].forEach(
			(letter, index) => {
				expect(wrapper.find('div.row-1').childAt(index).props().letter).toBe(
					letter
				);
			}
		);

		['A', 'S', 'D', 'F', 'G', 'H', 'J', 'K', 'L'].forEach((letter, index) => {
			expect(wrapper.find('div.row-2').childAt(index).props().letter).toBe(
				letter
			);
		});

		['Z', 'X', 'C', 'V', 'B', 'N', 'M'].forEach((letter, index) => {
			expect(wrapper.find('div.row-3').childAt(index).props().letter).toBe(
				letter
			);
		});

		expect(wrapper.find('div.row-3').childAt(7).props().letter).toBe(
			'Backspace'
		);
		expect(wrapper.find('div.row-3').childAt(8).props().letter).toBe('Enter');
	});
});
