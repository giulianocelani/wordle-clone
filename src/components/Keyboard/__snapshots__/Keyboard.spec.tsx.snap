// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Keyboard renders correctly 1`] = `
<div
  className="flex-1 flex flex-col items-center py-5 mx-auto justify-end"
>
  <div
    className="flex flex-col space-y-2 items-center"
  >
    <div
      className="row-1 flex space-x-2 w-full items-center justify-center"
      key="0"
    >
      <Key
        key="0-key-0"
        letter="Q"
        onClick={[MockFunction]}
      />
      <Key
        key="0-key-1"
        letter="W"
        onClick={[MockFunction]}
      />
      <Key
        key="0-key-2"
        letter="E"
        onClick={[MockFunction]}
      />
      <Key
        key="0-key-3"
        letter="R"
        onClick={[MockFunction]}
      />
      <Key
        key="0-key-4"
        letter="T"
        onClick={[MockFunction]}
      />
      <Key
        key="0-key-5"
        letter="Y"
        onClick={[MockFunction]}
      />
      <Key
        key="0-key-6"
        letter="U"
        onClick={[MockFunction]}
      />
      <Key
        key="0-key-7"
        letter="I"
        onClick={[MockFunction]}
      />
      <Key
        key="0-key-8"
        letter="O"
        onClick={[MockFunction]}
      />
      <Key
        key="0-key-9"
        letter="P"
        onClick={[MockFunction]}
      />
    </div>
    <div
      className="row-2 flex space-x-2 w-full items-center justify-center"
      key="1"
    >
      <Key
        key="1-key-0"
        letter="A"
        onClick={[MockFunction]}
      />
      <Key
        key="1-key-1"
        letter="S"
        onClick={[MockFunction]}
      />
      <Key
        key="1-key-2"
        letter="D"
        onClick={[MockFunction]}
      />
      <Key
        key="1-key-3"
        letter="F"
        onClick={[MockFunction]}
      />
      <Key
        key="1-key-4"
        letter="G"
        onClick={[MockFunction]}
      />
      <Key
        key="1-key-5"
        letter="H"
        onClick={[MockFunction]}
      />
      <Key
        key="1-key-6"
        letter="J"
        onClick={[MockFunction]}
      />
      <Key
        key="1-key-7"
        letter="K"
        onClick={[MockFunction]}
      />
      <Key
        key="1-key-8"
        letter="L"
        onClick={[MockFunction]}
      />
    </div>
    <div
      className="row-3 flex space-x-2 w-full items-center justify-center"
      key="2"
    >
      <Key
        key="2-key-0"
        letter="Z"
        onClick={[MockFunction]}
      />
      <Key
        key="2-key-1"
        letter="X"
        onClick={[MockFunction]}
      />
      <Key
        key="2-key-2"
        letter="C"
        onClick={[MockFunction]}
      />
      <Key
        key="2-key-3"
        letter="V"
        onClick={[MockFunction]}
      />
      <Key
        key="2-key-4"
        letter="B"
        onClick={[MockFunction]}
      />
      <Key
        key="2-key-5"
        letter="N"
        onClick={[MockFunction]}
      />
      <Key
        key="2-key-6"
        letter="M"
        onClick={[MockFunction]}
      />
      <Key
        key="2-key-7"
        letter="Backspace"
        onClick={[MockFunction]}
      />
      <Key
        key="2-key-8"
        letter="Enter"
        onClick={[MockFunction]}
      />
    </div>
  </div>
</div>
`;
